echopype/echodata/zarr_combine.py

```diff
@@ -0,0 +1,742 @@
+        For the keys ``date_created``, ``conversion_time`` the values
+        are not required to be identical, rather their type must be identical.
+        """
+
+        # make sure all keys are identical (this should never be triggered)
+        if attr1.keys() != attr2.keys():
+            raise RuntimeError(
+                "The attribute keys amongst the ds lists are not the same, combine cannot be used!"
+            )
+
+        # make sure that all values are identical
+        numpy_keys = []
+        for key in attr1.keys():
+
+            if isinstance(attr1[key], np.ndarray):
+
+                numpy_keys.append(key)
+
+                if not np.allclose(attr1[key], attr2[key], rtol=1e-12, atol=1e-12, equal_nan=True):
+                    raise RuntimeError(
+                        f"The attribute {key}'s value amongst the ds lists are not the "
+                        f"same, combine cannot be used!"
+                    )
+            elif key in ["date_created", "conversion_time"]:
+
+                if not isinstance(attr1[key], type(attr2[key])):
+                    raise RuntimeError(
+                        f"The attribute {key}'s type amongst the ds lists "
+                        f"are not the same, combine cannot be used!"
+                    )
+
+            else:
+
+                if attr1[key] != attr2[key]:
+                    raise RuntimeError(
+                        f"The attribute {key}'s value amongst the ds lists are not the "
+                        f"same, combine cannot be used!"
+                    )
+
+        return numpy_keys
+
+    def _get_ds_info(self, ds_list: List[xr.Dataset], ed_name: Optional[str]) -> None:
+        """
+        Constructs useful dictionaries that contain information
+        about the dimensions of the Dataset. Additionally, collects
+        the attributes from each Dataset in ``ds_list`` and saves
+        this group specific information to the class variable
+        ``group_attrs``.
+
+        Parameters
+        ----------
+        ds_list: List[xr.Dataset]
+            The Datasets that will be combined
+        ed_name: str
+            The name of the EchoData group corresponding to the
+            Datasets in ``ds_list``
+
+        Notes
+        -----
+        This method creates the following class variables:
+        dims_df: pd.DataFrame
+            Dataframe with column as dim names, rows as the
+            different Datasets, and values as the length of the dimension
+        dims_sum: dict
+            Keys as the dimension name and values as the corresponding
+            sum of the lengths across all Datasets
+        dims_csum: dict
+            Keys as the dimension name and values as a dictionary of
+            the corresponding cumulative sum of the lengths across
+            all Datasets
+        dims_max: dict
+            Keys as the dimension name and values as the corresponding
+            maximum length across all Datasets
```

Comment:
I wonder if it is worthwhile to initialize these as None in `__init__`, just for readability. The docstring here is very useful, and I wonder these should be also -- actually, I always have the question of where docstring of class/instance attributes should go...

Reply:
> I wonder if it is worthwhile to initialize these as None in __init__, just for readability.

This is a good point, I think that is good coding practice. I will go ahead and add these to `__init__`

> The docstring here is very useful, and I wonder where these should be also -- actually, I always have the question of where docstring of class/instance attributes should go...

Yes, I am always unsure of where to put these types of docstring, so I usually just put them in the method that creates them. Should we leave them where they are right now?

Reply:
Right, maybe we can leave what you have here in this method since it is where things are actually happening.

Maybe in the `__init__` section you could add a 1 line comment to send people to this method to find the details?

Reply:
Sounds good, I will put a comment pointing to `_get_ds_info` in `__init__`.